File: py-RayTracer/rtweekend.py
"""Utility functions for the Ray Tracing in One Weekend series."""

from math import pi
import random

INFINITY = float('inf')
PI = pi

def degrees_to_radians(degrees):
    """Convert degrees to radians."""
    return degrees * PI / 180.0

def radians_to_degrees(radians):
    """Convert radians to degrees."""
    return radians * 180.0 / PI

def random_double():
    """Returns a random real in [0,1)."""
    return random.random()

def random_double_range(min_val, max_val):
    """Returns a random real in [min_val, max_val)."""
    return min_val + (max_val-min_val)*random_double()

File: go-RayTracer/rtweekend.go
// Package rtweekend provides utility functions and constants for ray tracing.
// This is a port of the C++ code from "Ray Tracing in One Weekend" series.
package main

import (
	"math"
	"math/rand"
)

// Constants
const (
	Infinity = math.MaxFloat64
	Pi       = math.Pi
)

// DegreesToRadians converts degrees to radians
func DegreesToRadians(degrees float64) float64 {
	return degrees * Pi / 180.0
}

// RandomDouble returns a random float64 in [0,1)
func RandomDouble() float64 {
	// Returns a random real in [0,1)
	return rand.Float64()
}

// RandomDoubleRange returns a random float64 in [min,max)
func RandomDoubleRange(min, max float64) float64 {
	// Returns a random real in [min,max)
	return min + (max-min)*RandomDouble()
}
